pipeline-runner/R/gem2s-2-load_user_files.R

```diff
@@ -234,3 +254,84 @@ format_annot <- function(annot_list) {
   rownames(annot) <- annot$input
   return(annot)
 }
+
+# Fix annotations makes annotations compatible between samples with different types.
+# The possible options for features_types at this stage are 0,1,2
+#  0 is SYMBOL/ SYMBOL
+#  1 is IDS/SYMBOL
+#  2 is IDS/IDS
+equalize_annotation_types <- function(annot_list, counts_list, features_types_list, samples) {
+  if (any(features_types_list == 2) && any(features_types_list == 0) && !any(features_types_list == 1)) stop("Incompatible features detected.")
+
+  if (any(features_types_list == 1) && (any(features_types_list == 2) || any(features_types_list == 0))) {
+    annots_with_ids <- unique(do.call("rbind", annot_list[features_types_list == 1]))
+
+    annots_with_ids <- annots_with_ids[!duplicated(annots_with_ids$input), ]
+
+    for (sample in samples) {
+      if (features_types_list[[sample]] == 0 || features_types_list[[sample]] == 2) {
+        sample_annot <- annot_list[[sample]]
+
+        # Try to replace input column (currently symbols) in sample_annot with ids from annots_with_ids
+        if (features_types_list[[sample]] == 0) {
+
+          matched_symbols_index <- match(sample_annot$input,annots_with_ids$name)
+          is_in_annot_list <- which(sample_annot$input %in% annots_with_ids$name)
+
+          sample_annot$input[is_in_annot_list] <- annots_with_ids$input[na.omit(matched_symbols_index)]
+
+          #This avoids duplicates after combining with the annotated df.
+          #Leads to a mismatch in genes between samples but it seems like the best solution
+          sample_annot$input <- make.unique(sample_annot$input)
+
+          #The counts have been loaded with gene symbols so we need to replace the rownames with the ids
+          rownames(counts_list[[sample]]) <- sample_annot$input
+        }
+
+        # Try to replace names column (currently ids) in sample_annot with symbols from annots_with_ids
+        if (features_types_list[[sample]] == 2) {
+          matched_ids_index <- match(sample_annot$input,annots_with_ids$input)
+          is_in_annot_list <- which(sample_annot$input %in% annots_with_ids$input)
+
+          sample_annot$name[is_in_annot_list] <- annots_with_ids$name[na.omit(matched_ids_index)]
```

Comment:
so in here you are not replacing all the names of the sample, just some of them, what happens to the other ones that are not in `is_in_annot_list` ?

Reply:
They will remain with the same name. There are no annotations for those, we can't replace the names with something else